blockchain/chain_sync/src/tipset_syncer.rs

```diff
@@ -1154,7 +1154,9 @@ async fn validate_tipset<DB: Blockstore + Store + Clone + Send + Sync + 'static,
     let full_tipset_key = full_tipset.key().clone();
 
     let mut validations = FuturesUnordered::new();
-    for b in full_tipset.into_blocks() {
+    let blocks = full_tipset.into_blocks();
+    let blocks_len = blocks.len();
```

Comment:
you're using it in only one place, I guess it should be fine to not introduce a separate variable and inline it

Reply:
It's about the ownership, the for loop claims ownership of `blocks`, or maybe I can move the logging line above

Reply:
Moving the log earlier would make sense; the less code, the better!

Reply:
Fixed